src/NuGet.Services.SearchService/Controllers/SearchController.cs

```diff
@@ -107,6 +107,37 @@ public async Task<HttpResponseMessage> GetStatusAsync(HttpRequestMessage request
             return await _searchService.V3SearchAsync(request);
         }
 
+        [HttpGet]
+        public async Task<AutocompleteResponse> AutocompleteAsync(
+            int skip = DefaultSkip,
+            int take = DefaultTake,
+            bool prerelease = false,
+            string semVerLevel = null,
+            string q = null,
+            string id = null,
+            bool debug = false)
+        {
+            await EnsureInitializedAsync();
+
+            // If only "id" is provided, find package versions. Otherwise, find package Ids.
+            var type = (q != null || id == null)
```

Comment:
Is this behavior consistent with existing implementation? e.g.

- if both are provided with real values
- if either is provided with empty string, e.g. `q=&id=newtonsoft.json`
- if neither is provided

Reply:
Yup:

* If both are provided with real values, we return package ids. [Link](https://api-v2v3search-1.nuget.org/autocomplete?q=newtonsoft.json&id=newtonsoft.json)
* If both are provided but "id" is empty, we return package ids. [Link](https://api-v2v3search-1.nuget.org/autocomplete?q=Test&id=)
* If both are provided but "q" is empty, we return package ids. [Link](https://api-v2v3search-1.nuget.org/autocomplete?q=&id=Test)
* If neither are provided, we return package ids. [Link](https://api-v2v3search-1.nuget.org/autocomplete)